network/test/echoengine.go

```diff
@@ -56,9 +56,9 @@ func (te *EchoEngine) SubmitLocal(event interface{}) {
 
 // Submit is implemented for a valid type assertion to Engine
 // any call to it fails the test
-func (te *EchoEngine) Submit(originID flow.Identifier, event interface{}) {
+func (te *EchoEngine) Submit(channel network.Channel, originID flow.Identifier, event interface{}) {
```

Comment:
Please add a test for both `echoengine` and `meshengine` ensuring that network layer passes the _correct_ channel name to the engines. Passing the channel name is a new feature and we should make sure that we have it covered within our unittests.

Reply:
Added checks to all existing tests :)